banyand/metadata/schema/group.go

```diff
@@ -87,6 +88,9 @@ func (e *etcdSchemaRegistry) DeleteGroup(ctx context.Context, group string) (boo
 }
 
 func (e *etcdSchemaRegistry) CreateGroup(ctx context.Context, group *commonv1.Group) error {
+	if group.UpdatedAt != nil {
```

Comment:
I suppose we can move `UpdatedAt` field assignment to the `create` and `update` methods

Reply:
Refection? Would you be able to elaborate more on this?

Reply:
```go
// TypeMeta defines the identity and type of an Event.
type TypeMeta struct {
	Name  string
	Group string
	Kind  Kind
}

// Metadata wrap dedicated serialized resource and its TypeMeta.
type Metadata struct {
	Spec Spec
	TypeMeta
}
```

Reflection is not necessary. In the `Metadata`, we've already known its type. Use `switch...case` is enough

Reply:
Type assertion is not a good idea that introduces cycle dependence to the codes. Even if they lay in the same package, it is not a good practice.
I suggest leaving it alone.